django/thunderstore/usermedia/models/usermedia.py

```diff
@@ -0,0 +1,90 @@
+import copy
+
+import ulid2
+from django.conf import settings
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+from thunderstore.core.mixins import TimestampMixin
+from thunderstore.core.types import UserType
+from thunderstore.core.utils import ChoiceEnum
+
+
+class UserMediaQueryset(models.QuerySet):
+    def expired(self):
+        return self.exclude(Q(expiry=None) | Q(expiry__gt=timezone.now()))
+
+    def active(self):
+        return self.exclude(Q(expiry__lte=timezone.now()))
+
+
+class UserMediaStatus(ChoiceEnum):
+    initial = "initial"
+    upload_created = "upload_initiated"
+    upload_error = "upload_error"
+    upload_complete = "upload_complete"
+    upload_aborted = "upload_aborted"
+
+
+class UserMedia(TimestampMixin, models.Model):
+    objects = UserMediaQueryset.as_manager()
+
+    owner = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="usermedia",
+        on_delete=models.PROTECT,
+        blank=True,
+        null=True,
+    )
+    filename = models.CharField(max_length=1024)
+    key = models.CharField(max_length=2048)
+    size = models.PositiveIntegerField()
+    uuid = models.UUIDField(default=ulid2.generate_ulid_as_uuid, primary_key=True)
+    prefix = models.CharField(blank=True, null=True, max_length=256)
```

Comment:
What's `prefix`? Maybe this should be commented or renamed to something more specific

Reply:
It's only used for generating the key. In theory it might not even need to exist as a db column, but might as well keep it there if the key needs to be re-computed for some reason. Calling it `key_prefix` wouldn't make sense, since to me it'd suggest the key would get it prefixed in addition to what is already in the key (which already includes the prefix). `location` is what django-storages refers to it as (and what our setting where it's derived from is called), but since the field is only ever used in a single place, a few lines below it's declaration, I'm not sure if it warrants creating a new migration just to rename. It's not like you're supposed to interact with the field anyway, could be considered a private variable if those were a thing in db models..

I'll probably leave a comment and rename it in whatever migration potentially happens in the future.